src/main/kotlin/com/sourcegraph/cody/agent/CodyAgentService.kt

```diff
@@ -142,8 +142,8 @@ class CodyAgentService(project: Project) : Disposable {
 
     @JvmStatic
     fun setAgentError(project: Project, errorMsg: String?) {
-      agentError.set(errorMsg)
-      project.let { CodyStatusService.resetApplication(it) }
+      val oldErrorMsg = agentError.getAndSet(errorMsg)
```

Comment:
That is the fix for the completions status, calling `resetApplication` was causing it to be reseted too quickly.

Reply:
wow, this is fine